conans/client/configure_build_environment.py

```diff
@@ -89,11 +92,89 @@ def __init__(self, conanfile):
 
+    def _get_host_build_target_flags(self, arch_detected, os_detected):
+        """Based on google search for build/host triplets, it could need a lot and complex verification"""
+        if not cross_building(self._conanfile.settings, os_detected, arch_detected):
+            return None, None, None
+
+        arch_setting = self._conanfile.settings.get_safe("arch")
+        os_setting = self._conanfile.settings.get_safe("os")
+        os_detected = os_detected
+
+        if os_detected == "Windows" and os_setting != "Windows":
+            return None, None, None    # Don't know what to do with these, even exists? its only for configure
+
+        # Building FOR windows
+        if os_setting == "Windows":
+            build = "i686-w64-mingw32" if arch_detected == "x86" else "x86_64-w64-mingw32"
+            host = "i686-w64-mingw32" if arch_setting == "x86" else "x86_64-w64-mingw32"
+        else:  # Building for Linux or Android
+            build = "%s-%s" % (arch_detected, {"Linux": "linux-gnu", "Darwin": "apple-macos"}.get(os_detected,
+                                                                                                  os_detected.lower()))
+            if arch_setting == "armv8":
+                host_arch = "aarch64"
+            else:
+                host_arch = "arm" if "arm" in arch_setting else arch_setting
+
+            host = "%s%s" % (host_arch, {"Linux": "-linux-gnueabi",
+                                         "Android": "-linux-android"}.get(os_setting, ""))
+            if arch_setting == "armv7hf" and os_setting == "Linux":
+                host += "hf"
+            elif "arm" in arch_setting and arch_setting != "armv8" and os_setting == "Android":
+                host += "eabi"
+
+        return build, host, None
+
+    def configure(self, configure_dir=None, args=None, build=None, host=None, target=None):
+        """
+        :param configure_dir: Absolute or relative path to the configure script
+        :param args: Optional arguments to pass to configure.
+        :param build: In which system the program will be built. "False" skips the --build flag
+        :param host: In which system the generated program will run.  "False" skips the --host flag
+        :param target: This option is only used to build a cross-compiling toolchain.  "False" skips the --target flag
+                       When the tool chain generates executable program, in which target system the program will run.
+        :return: None
+
+        http://jingfenghanmax.blogspot.com.es/2010/09/configure-with-host-target-and-build.html
+        https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html
+
+        """
+        configure_dir = configure_dir or "./"
+        auto_build, auto_host, auto_target = None, None, None
+        if build is None or host is None or target is None:
+            auto_build, auto_host, auto_target = self._get_host_build_target_flags(detected_architecture(), platform.system())
```

Comment:
A bit spaghetti, you are passing detected_architecture() as param, but this will be the default of ``cross_building()`` if no value is provided. Same for platform.system(). Please check.